libbeat/outputs/elasticsearch/client.go

```diff
@@ -445,7 +447,8 @@ func (client *Client) bulkCollectPublishFails(result eslegclient.BulkResult, dat
 					}
 				} else { // drop
 					stats.nonIndexable++
-					client.log.Warnf("Cannot index event %#v (status=%v): %s, dropping event!", data[i], status, msg)
+					client.log.Warnf("Cannot index event (status=%v): dropping event!", status)
+					client.log.Debugf("Cannot index event %#v (status=%v): %s, dropping event!", data[i], status, msg)
```

Comment:
Are we able to safely log the `msg`? Does it contain event data if you simulate a mapping conflict (map something as keyword and then write a json object to it or something similar)? I would hope just the field names involved.

Reply:
If it is just the fields names then I guess we can remove the debug log and only log the msg right?

Reply:
Unfortunately Elasticsearch returns both the field name and its value, I updated the PR description with a detailed reasoning, but here is the TLDR:

Given the event:
```
{"message":"index failure","int":"not a number","string":10}
```

The field `int` is supposed to be an number (see the "how to test this PR above"), that causes the mapping failure with the following message:
```
{
  "type": "document_parsing_exception",
  "reason": "[1:437] failed to parse field [int] of type [long] in document with id 'CdYVHIwBAsOaLnJkwT8o'. Preview of field's value: 'not a number'",
  "caused_by": {
    "type": "illegal_argument_exception",
    "reason": "For input string: \"not a number\""
  }
}
```

Reply:
We have seen instances of payload events showing up on the RLC logs on Synthetics service when there were mapping conflicts (Mostly likely customer has modified mappings on their ES stack). Especially happened for some of the network payload events which had cookies/auth headers which shouldn't be logged.

Reply:
Yeah that is unfortunate. We may be able to parse these and log the pieces that don't contain the field values but that has to be a separate change from this I think.

Reply:
@belimawr I agree with what Craig just said above. Let's change the log level for now but remove the mention of `Preview of field's value: 'xxx'"` in a follow up PR.

Reply:
To be clear - we can continue with this PR as written to get a fix in the short-term, but I do think losing the error message is going to make debugging issues quite painful. We should have Elasticsearch change this (or add a different field with just the error message without the field value) and then come back and log this information again when it's sure it doesn't contain sensitive information.

Reply:
Agreed, there is already a follow up issue in the security repository about having ES make a change here as well.

Reply:
I added @cmacknz's suggestions, the way it is now:
1. The original log entries are in debug level
2. For all entries I changed the level, there is a shorter one in the original log level stating that more details are available in debug level.